examples/language-modeling/run_language_modeling.py

```diff
@@ -33,6 +33,7 @@
     AutoModelWithLMHead,
     AutoTokenizer,
     DataCollatorForLanguageModeling,
+    DataCollatorForXLNetLanguageModeling,
```

Comment:
A part of me thinks it may not be a good design. Maybe you should instead implement Permutated LM objective in the `DataCollatorForLanguageModeling` since it currently supports MLM and traditional LM (i.e., autoregressive) objective.

Reply:
I agree; that would be a cleaner solution. Should I create a separate `--plm` flag like `--mlm` and add a new function `mask_tokens_and_perm()` like `mask_tokens()` in `DataCollatorForLanguageModeling`?

Reply:
@patrickvonplaten @LysandreJik What do you think?

Reply:
I actually prefer the way it is done now (Creating a completely new data collator for XLNet). Also pinging @julien-c to hear his opinion on this